cmd/src/gateway_benchmark.go

```diff
@@ -0,0 +1,258 @@
+package main
+
+import (
+	"crypto/tls"
+	"flag"
+	"fmt"
+	"io"
+	"net/http"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/sourcegraph/src-cli/internal/cmderrors"
+)
+
+const (
+	defaultRequestCount = 100
+	blue                = "\033[34m"
+	green       = "\033[32m"
+	yellow      = "\033[33m"
+	red         = "\033[31m"
+	reset       = "\033[0m"
+)
+
+func init() {
+	usage := `
+'src gateway benchmark' runs performance benchmarks against Cody Gateway endpoints.
+
+Usage:
+
+    src gateway benchmark [flags]
+
+Examples:
+
+    $ src gateway benchmark
+    $ src gateway benchmark --requests 50
+`
+
+	flagSet := flag.NewFlagSet("benchmark", flag.ExitOnError)
+
+	var (
+		requestCount = flagSet.Int("requests", defaultRequestCount, "Number of requests to make per endpoint")
+	)
+
+	handler := func(args []string) error {
+		if err := flagSet.Parse(args); err != nil {
+			return err
+		}
+
+		if len(flagSet.Args()) != 0 {
+			return cmderrors.Usage("additional arguments not allowed")
+		}
+
+		// Create HTTP client with TLS skip verify
+		client := &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+			},
+		}
+
+		endpoints := map[string]string{
+			"HTTP":                fmt.Sprintf("%s/cody-gateway-call-http", cfg.Endpoint),
+			"HTTP then WebSocket": fmt.Sprintf("%s/cody-gateway-call-http-then-websocket", cfg.Endpoint),
+		}
+
+		fmt.Printf("Starting benchmark with %d requests per endpoint...\n", *requestCount)
+
+		var results []endpointResult
+
+		for name, url := range endpoints {
+			durations := make([]time.Duration, 0, *requestCount)
+			fmt.Printf("\nTesting %s...", name)
+
+			for i := 0; i < *requestCount; i++ {
+				duration := benchmarkEndpoint(client, url)
+				if duration > 0 {
+					durations = append(durations, duration)
+				}
+				fmt.Printf("\rTesting %s: %d/%d", name, i+1, *requestCount)
+			}
+			fmt.Println()
+
+			avg, p5, p95, median, total := calculateStats(durations)
```

Comment:
let's make calculateStats return a struct with these as fields, so there's no chance we get the order wrong here and look at the wrong thing by accident.